pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -656,6 +671,9 @@ func (oe *operationExecutor) generateMountVolumeFunc(
 		mountErr := volumeMounter.SetUp(fsGroup)
 		if mountErr != nil {
 			// On failure, return error. Caller will log and retry.
+			if errGetRef == nil && ref != nil {
```

Comment:
Is it possible to fire an event without a reference? If so, I wonder if that would be worth it even if we failed to get the reference.